### PR TITLE
Agoric Support

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Add supports for Agoric blockchain
https://docs.agoric.com

## How to test
Unit tests are written in chain specific folder

Transaction hash for signer test
https://atomscan.com/agoric/transactions/142BF63F54D059D8B3C356F6E6F36EA3A979AEE6A0875E0F350D71F3799B030F

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Comment by catenocrypt** [comment]
Thank you for this contribution too. We will check if this project matches the criteria to be integrated at this point.

File: docs/registry.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ This list is generated from [./registry.json](../registry.json)
 | 501     | Solana           | SOL    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/solana/info/logo.png" width="32" />       | <https://solana.com>          |
 | 508     | Elrond           | eGLD   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/elrond/info/logo.png" width="32" />       | <https://elrond.com/>         |
 | 529     | Secret           | SCRT   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/secret/info/logo.png" width="32" />       | <https://scrt.network/>       |
+| 564     | Agoric           | BLD    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/agoric/info/logo.png" width="32" />       | <https://agoric.com>          |
```
Comment (catenocrypt) [inline docs/registry.md:54]: The logo is missing from `assets`, could you create PR there withe the logo&info?

File: registry.json [hunk 2]
```diff
@@ -2880,5 +2880,37 @@
       "source": "https://github.com/hashgraph",
       "documentation": "https://docs.hedera.com"
     }
+  },
+  {
+    "id": "agoric",
+    "name": "Agoric",
+    "displayName": "Agoric",
+    "coinId": 564,
+    "symbol": "BLD",
```
Comment (catenocrypt) [inline registry.json:2889]: According to https://github.com/satoshilabs/slips/blob/master/slip-0044.md , symbol is 'AG', please clarify.

**Review by catenocrypt — DISMISSED** [review]

File: registry.json (hunk 2 above)
Comment (AyushBherwani1998) [reply registry.json:2889]: The information was taken from https://github.com/cosmos/chain-registry/blob/12a99dc31656b767fabe474f01468f6d2bc62c36/agoric/assetlist.json#L6

There is no coin with symbol `AG` in Agoric.

File: docs/registry.md (hunk 1 above)
Comment (AyushBherwani1998) [reply docs/registry.md:54]: I have created a new PR.

File: registry.json [hunk 3]
```diff
@@ -2880,5 +2880,37 @@
       "source": "https://github.com/hashgraph",
       "documentation": "https://docs.hedera.com"
     }
+  },
+  {
+    "id": "agoric",
+    "name": "Agoric",
+    "displayName": "Agoric",
```
Comment (hewigovens) [inline registry.json:2887]: This can be omitted if same as `name`

**Review by hewigovens — APPROVED** [review]

File: registry.json (hunk 3 above)
Comment (AyushBherwani1998) [reply registry.json]: @hewigovens done

**Review by hewigovens — APPROVED** [review]

**Review by Milerius — APPROVED** [review]

**Comment by Milerius** [comment]
Thanks for the contribution @AyushBherwani1998 !

**Pull request merged**